File: gen_scripts/python2/j11.arguments.py
#!/usr/bin/env python
import sys
# illustrative variables 
ARG_COUNT = len(sys.argv) - 1 # get num of real arguments
FIRST     = 1                 # index of first argument

print "The arugments passed are:"
# use collection loop with range to enumerate arguments
#   Note: range ending must be one greater than desired range
for count in range(FIRST,ARG_COUNT+1): # range to gen sequence of numbers
   arg = sys.argv[count]               # get arg using index
   # print count and argument using count index
   print " item %d: %s" % (count,arg) 

File: gen_scripts/perl/j11.arguments.pl
#!/usr/bin/env perl -w
# illustrative variables
my $last  = $#ARGV; # get index of last element
my $first = 0;      # set index of first element

print "The arguments passed are:\n";
# collection loop with range to enumerate args
#   Notes:
#    - $count     is index into array
#    - $count + 1 is count that is displayed to user
foreach $count ($first..$last) {
  my $arg = $ARGV[$count];  # get arg using index
  # output count and argument using count index
  printf " item %d: %s\n", $count+1, $arg;
}
